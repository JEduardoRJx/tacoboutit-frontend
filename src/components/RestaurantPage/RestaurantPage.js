import React, { Component } from 'react'
import { Text, View, Image, StyleSheet, Dimensions, Linking, TouchableOpacity, ScrollView, Modal } from 'react-native';
import { LinearGradient } from "expo-linear-gradient";
import TacoCard from '../TacoCard/TacoCard';
import AddTacoButton from '../AddTacoButton/AddTacoButton';
import ReviewsPage from '../ReviewsPage/ReviewsPage'
const callIcon = require('../../../assets/call-answer.png');

export default class RestaurantPage extends Component { 
  state = {
    showReviews: false,
  }

  renderTacos = () => {
    const { tacos } = this.props.restaurant;
    return tacos.map((taco, i) => <TacoCard key={`${taco.type}-${i}`} type={taco.type} />);
  }

  render() {
    const { restaurant, submitTaco, updateLocalReviews } = this.props;
    const id = restaurant.id;
    const tacos = restaurant.tacos;
    const status = restaurant.isClosed ? 'Closed' : 'Open';
    const color = restaurant.isClosed ? 'red': 'green';
    return (
      <LinearGradient
        colors={["#F0CB35", "#D56C2C", "#C02425"]}
        style={styles.container}>
        <View style={styles.statusBubble}>
            <Text style={{ color, textAlign: 'center', fontSize: 20, fontWeight: 'bold' }}>{status}</Text>
        </View>
        <AddTacoButton style={styles.addTacoButton} submitTaco={submitTaco} id={id}/>
        <Image
            style={styles.img}
            source={{ uri: restaurant.image_url }}/>
        <ScrollView>
          <View style={styles.restaurantInfoContainer}>
            <Text style={styles.header}>{restaurant.name}</Text>
            <Text style={styles.restaurantInfo}>{restaurant.address}</Text>
            <TouchableOpacity 
              onPress={() => Linking.openURL(`tel:${restaurant.phone}`)}
              style={styles.phoneBubble}
            >
              <Image
                style={{width: 40, height: 40}}
                source={callIcon}
              />
            </TouchableOpacity>
            {tacos.length !== 0 && <View>{this.renderTacos()}</View>}
            <TouchableOpacity onPress={() => this.setState({showReviews: true})}>
                <Modal visible={this.state.showReviews}
                  onRequestClose={() => this.setState({showReviews: false})}>
                  <ReviewsPage tacos={tacos} updateLocalReviews={updateLocalReviews} />
                </Modal>
                <View style={{justifyContent: 'center', alignItems: 'center', width: '100%'}}>
                  <Text style={styles.viewAllReviewsStyles}>View All Reviews</Text>
                </View>
              </TouchableOpacity>
          </View>
        </ScrollView>
      </LinearGradient>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  img: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height * .4
  },
  restaurantInfoContainer: {
    flex: 2,
    padding: 10,
  },
  header: {
    color: 'white',
    fontSize: 36,
    fontWeight: 'bold',
    textAlign: 'left',
  },
  restaurantInfo: {
    fontSize: 22,
    color: 'white',
  },
  statusBubble: {
    width: 100,
    height: 40,
    backgroundColor: 'white',
    borderRadius: 20,
    justifyContent: 'center',
    position: 'absolute',
    zIndex: 5,
    top: 10,
    right: 10,
  },
  phoneBubble: {
    backgroundColor: 'white', 
    width: Dimensions.get('window').width * .8, 
    height: 60, 
    justifyContent: 'center', 
    alignItems: 'center', 
    borderRadius: 25,
    alignSelf: 'center',
    marginVertical: 5,
  },
  viewAllReviewsStyles: {
    backgroundColor: '#00BFFF',
    color: 'white',
    height: 50,
    width: '70%',
    borderRadius: 30, 
    textAlign: 'center',
    fontSize: 24,
    textAlignVertical: 'center'
  }
})