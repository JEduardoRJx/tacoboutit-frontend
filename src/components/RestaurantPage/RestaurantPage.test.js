import React from 'react';
import renderer from 'react-test-renderer';

import RestaurantPage from '../RestaurantPage/RestaurantPage';

describe('RestaurantPage', () => {
  const mockRestaurant = {
    address: "2835 W 72nd Ave, Westminster, CO 80030",
    distance: 3.52669047076749,
    id: 1,
    image_url: "https://s3-media2.fl.yelpcdn.com/bphoto/tNN5NvMch7rJMm-YyTLTxA/o.jpg",
    is_closed: false,
    latitude: 39.82754,
    longitude: -105.02161,
    name: "Guadalajara Mexican Restaurant",
    phone: "+13034269540",
    review_count: 471,
    tacoboutit_item_review_count: 0,
    tacos: [{
        id: 1,
        restaurant: 1,
        type: "al pastor",
      },
       {
        id: 2,
        restaurant: 1,
        type: "carne asada",
      },
      {
        id: 3,
        restaurant: 1,
        type: "cabeza",
      },
       {
        id: 8,
        restaurant: 1,
        type: "testing",
      },
      {
        id: 13,
        restaurant: 1,
        type: "carnitas",
      },
    ],
    url: "https://www.yelp.com/biz/guadalajara-mexican-restaurant-westminster?adjust_creative=pxQ3XEqH9sM15FQYWsuBXQ&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=pxQ3XEqH9sM15FQYWsuBXQ",
    yelp_id: "3LoMv9TtprQhH1AmN33ubA",
    yelp_rating: 4.5,
  }

  it('renders correctly', () => {
    const wrapper = renderer.create(<RestaurantPage restaurant={mockRestaurant} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});