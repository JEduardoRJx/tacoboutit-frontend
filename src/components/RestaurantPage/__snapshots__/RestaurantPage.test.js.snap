// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantPage renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4293970741,
        4292176940,
        4290782245,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "position": "absolute",
        "right": 10,
        "top": 10,
        "width": 100,
        "zIndex": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "green",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Open
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 10,
        "top": 220,
        "width": 100,
        "zIndex": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      + Taco
    </Text>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DCDCDC",
            "borderRadius": 20,
            "height": 667,
            "justifyContent": "space-around",
            "left": 75,
            "overflow": "hidden",
            "padding": 20,
            "position": "absolute",
            "top": 333.5,
            "width": 600,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Add New Taco
        </Text>
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Select Type
        </Text>
        <View
          style={
            Object {
              "height": 50,
              "width": "100%",
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Carne Asada",
                  "textColor": undefined,
                  "value": "carne asada",
                },
                Object {
                  "label": "Al Pastor",
                  "textColor": undefined,
                  "value": "al pastor",
                },
                Object {
                  "label": "Cabeza",
                  "textColor": undefined,
                  "value": "cabeza",
                },
                Object {
                  "label": "Cabeza",
                  "textColor": undefined,
                  "value": "cabeza",
                },
                Object {
                  "label": "Carnitas",
                  "textColor": undefined,
                  "value": "carnitas",
                },
                Object {
                  "label": "Vegan",
                  "textColor": undefined,
                  "value": "vegan",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00BFFF",
                "borderRadius": 30,
                "height": 50,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                }
              }
            >
              Submit
            </Text>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <Image
    source={
      Object {
        "uri": "https://s3-media2.fl.yelpcdn.com/bphoto/tNN5NvMch7rJMm-YyTLTxA/o.jpg",
      }
    }
    style={
      Object {
        "height": 533.6,
        "width": 750,
      }
    }
  />
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flex": 2,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 36,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Guadalajara Mexican Restaurant
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 22,
            }
          }
        >
          2835 W 72nd Ave, Westminster, CO 80030
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 25,
              "height": 60,
              "justifyContent": "center",
              "marginVertical": 5,
              "opacity": 1,
              "width": 600,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          />
        </View>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "white",
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 10,
                "padding": 10,
                "width": 562.5,
              }
            }
          >
            <Text>
              al pastor
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "white",
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 10,
                "padding": 10,
                "width": 562.5,
              }
            }
          >
            <Text>
              carne asada
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "white",
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 10,
                "padding": 10,
                "width": 562.5,
              }
            }
          >
            <Text>
              cabeza
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "white",
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 10,
                "padding": 10,
                "width": 562.5,
              }
            }
          >
            <Text>
              testing
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "white",
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 10,
                "padding": 10,
                "width": 562.5,
              }
            }
          >
            <Text>
              carnitas
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Modal
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            visible={false}
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4293970741,
                    4292176940,
                    4290782245,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": undefined,
                }
              }
              style={
                Object {
                  "height": 1334,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 9,
                  }
                }
              >
                <RCTScrollView>
                  <View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgba(225, 225, 225, 0.3)",
                          "borderRadius": 20,
                          "margin": 20,
                          "marginTop": 15,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.3)",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            AL PASTOR
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            <Text>
                              5
                              /10 Avg.
                            </Text>
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "padding": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
                              "borderBottomWidth": 2,
                              "borderRadius": 30,
                              "marginBottom": 20,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            4
                            /10
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            Great spot
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgba(225, 225, 225, 0.3)",
                          "borderRadius": 20,
                          "margin": 20,
                          "marginTop": 15,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.3)",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            CARNE ASADA
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            <Text>
                              5
                              /10 Avg.
                            </Text>
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "padding": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
                              "borderBottomWidth": 2,
                              "borderRadius": 30,
                              "marginBottom": 20,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            4
                            /10
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            Great spot
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgba(225, 225, 225, 0.3)",
                          "borderRadius": 20,
                          "margin": 20,
                          "marginTop": 15,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.3)",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            CABEZA
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            <Text>
                              5
                              /10 Avg.
                            </Text>
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "padding": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
                              "borderBottomWidth": 2,
                              "borderRadius": 30,
                              "marginBottom": 20,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            4
                            /10
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            Great spot
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgba(225, 225, 225, 0.3)",
                          "borderRadius": 20,
                          "margin": 20,
                          "marginTop": 15,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.3)",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            TESTING
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            <Text>
                              5
                              /10 Avg.
                            </Text>
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "padding": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
                              "borderBottomWidth": 2,
                              "borderRadius": 30,
                              "marginBottom": 20,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            4
                            /10
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            Great spot
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgba(225, 225, 225, 0.3)",
                          "borderRadius": 20,
                          "margin": 20,
                          "marginTop": 15,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.3)",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            CARNITAS
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "flex": 1,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "justifyContent": "space-between",
                                "textAlign": "center",
                              }
                            }
                          >
                            <Text>
                              5
                              /10 Avg.
                            </Text>
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "padding": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
                              "borderBottomWidth": 2,
                              "borderRadius": 30,
                              "marginBottom": 20,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            4
                            /10
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            Great spot
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {
                      "paddingBottom": 0,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#00BFFF",
                      "flex": 1,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 32,
                      }
                    }
                  >
                    Add Review
                  </Text>
                </View>
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
          </Modal>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "#00BFFF",
                  "borderRadius": 30,
                  "color": "white",
                  "fontSize": 24,
                  "height": 50,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "70%",
                }
              }
            >
              View All Reviews
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</ViewManagerAdapter_ExpoLinearGradient>
`;
